pkg/cli/cmd/credential/register/azure/wi/workloadidentity.go

```diff
@@ -0,0 +1,180 @@
+to configure these settings.
+
+Radius will use the provided workload identity credential for all interactions with Azure, including Bicep deployment, 
+Radius Environments, and Radius portable resources.
+
+Radius will use the provided subscription and resource group as the default target scope for Bicep deployment.
+The provided service principal must have the Contributor or Owner role assigned for the provided resource group
+in order to create or manage resources contained in the group. The resource group should be created before
+calling 'rad credential register azure wi'.
+` + common.LongDescriptionBlurb,
+		Example: `
+# Register (Add or update) cloud provider credential for Azure with workload identity authentication
+rad credential register azure wi --client-id <client id/app id> --tenant-id <tenant id>
+`,
+		Args: cobra.ExactArgs(0),
+		RunE: framework.RunCommand(runner),
+	}
+
+	commonflags.AddOutputFlag(cmd)
+	commonflags.AddWorkspaceFlag(cmd)
+
+	cmd.Flags().String("client-id", "", "The client id or app id of an Azure service principal.")
+	_ = cmd.MarkFlagRequired("client-id")
+
+	cmd.Flags().String("tenant-id", "", "The tenant id of an Azure service principal.")
+	_ = cmd.MarkFlagRequired("tenant-id")
+
+	return cmd, runner
+}
+
+// Runner is the runner implementation for the `rad credential register azure wi` command.
+type Runner struct {
+	ConfigHolder      *framework.ConfigHolder
+	ConnectionFactory connections.Factory
+	Output            output.Interface
+	Format            string
+	Workspace         *workspaces.Workspace
+
+	ClientID    string
+	TenantID    string
+	KubeContext string
+}
+
+// NewRunner creates a new instance of the `rad credential register azure wi` runner.
+func NewRunner(factory framework.Factory) *Runner {
+	return &Runner{
+		ConfigHolder:      factory.GetConfigHolder(),
+		ConnectionFactory: factory.GetConnectionFactory(),
+		Output:            factory.GetOutput(),
+	}
+}
+
+// Validate runs validation for the `rad credential register azure wi` command.
+//
+
+// Validate checks for the presence of a workspace, output format, client ID, and tenant ID, and
+// sets them in the Runner struct if they are present. If any of these are not present, an error is returned.
+func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
+	workspace, err := cli.RequireWorkspace(cmd, r.ConfigHolder.Config, r.ConfigHolder.DirectoryConfig)
+	if err != nil {
+		return err
+	}
+	r.Workspace = workspace
+
+	format, err := cli.RequireOutput(cmd)
+	if err != nil {
+		return err
+	}
+	r.Format = format
+
+	clientID, err := cmd.Flags().GetString("client-id")
+	if err != nil {
+		return err
+	}
+
+	tenantID, err := cmd.Flags().GetString("tenant-id")
+	if err != nil {
+		return err
+	}
+
+	r.ClientID = clientID
+	r.TenantID = tenantID
+
+	kubeContext, ok := r.Workspace.KubernetesContext()
+	if !ok {
+		return clierrors.Message("A Kubernetes connection is required.")
+	}
+	r.KubeContext = kubeContext
```

Comment:
Do we need the kubecontext? looks unused

Reply:
removed